packages/ui-components/src/components/windowedlist.ts

```diff
@@ -1524,16 +1692,56 @@ export namespace WindowedList {
    * Windowed list view constructor options
    */
   export interface IOptions<
-    T extends WindowedList.IModel = WindowedList.IModel
+    T = any,
+    U extends WindowedList.IModel<T> = WindowedList.IModel<T>
   > {
     /**
      * Windowed list model to display
      */
-    model: T;
+    model: U;
     /**
      * Windowed list layout
      */
     layout?: WindowedLayout;
+
+    /**
+     * A renderer for the elements of the windowed list.
+     */
+    renderer?: IRenderer<T>;
+
+    /**
+     * Whether the windowed list should display a scrollbar UI.
+     */
+    scrollbar?: boolean;
+  }
+
+  /**
+   * A windowed list element renderer.
+   */
+  export interface IRenderer<T = any> {
+    /**
+     * Create the outer, root element of the windowed list.
+     */
+    createOuter(): HTMLElement;
+
+    /**
+     * Create the virtual scrollbar element.
+     */
+    createScrollbar(): HTMLOListElement;
```

Comment:
I'm not sure about this because it is inherently an ordered list (going into the windowed list) ... and both an `<ol>` and its children `<li>` elements are very flexible in the content they support. I would propose keeping this signature to give consumers of it some idea and structure for what the underlying code is doing.